### PR TITLE
PHPC-1960: Do not use BulkWriteException for client-side errors

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPC-1960

### CODE REVIEW

File: tests/manager/manager-executeBulkWrite_error-005.phpt [hunk 1]
```diff
@@ -24,45 +24,14 @@ $bulk->insert(['x' => 1]);
...
-  NULL
-  ["writeConcern"]=>
-  object(MongoDB\Driver\WriteConcern)#%d (%d) {
-  }
-  ["errorReplies"]=>
-  array(0) {
-  }
-}
+OK: Got MongoDB\Driver\Exception\ConnectionTimeoutException
+Failed to send "delete" command with database "%s": Failed to read 4 bytes: socket error or timeout
```
Comment (GromNaN) [inline tests/manager/manager-executeBulkWrite_error-005.phpt:36]: The current result is:
```
Deprecated: Creation of dynamic property MongoDB\Driver\Exception\RuntimeException::$writeResult is deprecated in /Users/jerome/Develop/mongo-php-driver/tests/manager/manager-executeBulkWrite_error-005.php on line 20
ALMOST: Got MongoDB\Driver\Exception\RuntimeException - expected MongoDB\Driver\Exception\ConnectionTimeoutException
Failed to send "delete" command with database "phongo": Failed to read 4 bytes: socket error or timeout
```

I need to fix the "dynamic property" issue and the exception class.

Comment (GromNaN) [reply tests/manager/manager-executeBulkWrite_error-005.phpt:36]: I got the correct exception class by using `phongo_throw_exception_from_bson_error_t_and_reply(&error, &reply)`.

```
Deprecated: Creation of dynamic property MongoDB\Driver\Exception\ConnectionTimeoutException::$writeResult is deprecated in /Users/jerome/Develop/mongo-php-driver/tests/manager/manager-executeBulkWrite_error-005.php on line 20
OK: Got MongoDB\Driver\Exception\ConnectionTimeoutException
Failed to send "delete" command with database "phongo": Failed to read 4 bytes: socket error or timeout
```

Now I need to figure where the dynamic property comes from.

Comment (GromNaN) [reply tests/manager/manager-executeBulkWrite_error-005.phpt:36]: Test successful after adding `ConnectionTimeoutException::$writeResult`.

**Comment by jmikola** [comment]
Closed [PHPC-1960](https://jira.mongodb.org/browse/PHPC-1960) as "Won't Do", so closing this as well. See that issue for an explanation.

**Pull request closed without merging**